src/core/dash.h

```diff
@@ -960,12 +967,36 @@ auto DashTable<_Key, _Value, Policy>::Traverse(Cursor curs, Cb&& cb) -> Cursor {
 
 template <typename _Key, typename _Value, typename Policy>
 template <typename Cb>
-void DashTable<_Key, _Value, Policy>::TraverseBucket(const_iterator it, Cb&& cb) {
-  SegmentType& s = *segment_[it.seg_id_];
-  const auto& b = s.GetBucket(it.bucket_id_);
-  b.ForEachSlot([it, cb = std::move(cb), table = this](auto* bucket, uint8_t slot, bool probe) {
-    cb(iterator{table, it.seg_id_, it.bucket_id_, slot});
-  });
+auto DashTable<_Key, _Value, Policy>::TraverseBuckets(Cursor curs, Cb&& cb) -> Cursor {
+  if (curs.bucket_id() >= SegmentType::kTotalBuckets)  // sanity.
+    return 0;
+
+  uint32_t sid = curs.segment_id(global_depth_);
+  uint8_t bid = curs.bucket_id();
+
+  bool fetched = false;
```

Comment:
Nit: While it's not strictly necessary to iterate until we find a non-empty bucket, we could potentially iterate over the entire empty table without returning. Conversely, for densely packed tables, we might return after checking each bucket. To balance these scenarios, iterating over a fixed number of buckets (e.g., 8 or 16) regardless of the number of non-empty buckets encountered might be a smoother approach.